src/geo/transform.js

```diff
@@ -1599,7 +1599,7 @@ class Transform {
         const pos = this._computeCameraPosition(mercPixelsPerMeter);
 
         const elevationAtCamera = elevation.getAtPointOrZero(new MercatorCoordinate(...pos));
-        const minHeight = this._minimumHeightOverTerrain() * Math.cos(degToRad(this._maxPitch));
+        const minHeight = this._minimumHeightOverTerrain() * Math.cos(degToRad(85));
```

Comment:
What's the reason for hard-coding this value? Maybe add context in a comment?

Reply:
Added a comment. In maps with maxPitch set to less than 85, fast-zooming past terrain is exasperated since it would adjust the camera position to early. Hard-coding this value to 85 instead of maxPitch reduces this experience in these cases

Reply:
Should this be `Math.min(this._maxPitch, 85)` so that it works properly with explicitly set lower value?

Reply:
Since `Math.cos(degToRad(85))` < `Math.cos(degToRad(0))`, should we use the upper bound of the possible values in order to keep a constant `minHeigh` safe bound? That would mean directly using:
```
        const minHeight = this._minimumHeightOverTerrain();
```
Since `Math.cos(degToRad(0))` is 1. Would that work or lead to issues in practice?

Reply:
The initial experience of using `minHeight = this._minimumHeightOverTerrain()` is a bit more jarring. if you bounce into the terrain it pushes the camera farther up and away. It also allowed for the camera to go under the terrain. I'll look into why

Reply:
Nit: If this is the correct value, should we consider hard coding the result in order to avoid an `unnecessary cos()` operation? 
```suggestion
        const cosMaxPitch = 0.08715560745961583
        const minHeight = this._minimumHeightOverTerrain() * cosMaxPitch;
```

Though, have you experimented with other values to see if you the behavior can be smoother?